Lib/uuid.py

```diff
@@ -700,24 +728,31 @@ def uuid3(namespace, name):
     """Generate a UUID from the MD5 hash of a namespace UUID and a name."""
     if isinstance(name, str):
         name = bytes(name, "utf-8")
-    from hashlib import md5
-    digest = md5(
-        namespace.bytes + name,
-        usedforsecurity=False
-    ).digest()
-    return UUID(bytes=digest[:16], version=3)
+    import _md5
+    h = _md5.md5(namespace.bytes + name, usedforsecurity=False)
+    assert len(h.digest()) == 16
```

Comment:
Also this takes time, in particular the `.digest()`

Reply:
Ah right, I generally use `python -OO` when I'm dealing with performance-critical objects

Reply:
I've removed the assert altoghether. If the MD5 digest is not 16 bytes, something is fundamentally wrong elsewhere.